### DISCUSSION TITLE
Numbers extend beyond the edge of the table cells [solved]

### DISCUSSION THREAD

**Discussion by sindzicat** [lvjr/tabularray#505] [Q&A]
Hello all!

```latex
\documentclass{article}

\usepackage{xcolor}
\usepackage{tabularray}
\UseTblrLibrary{siunitx}

\begin{document}
    \begin{tblr}{
        width=\linewidth,
        hlines = {gray},
        vlines={dashed},
         colspec={
                 Q[si={table-format=+1.2e3,table-number-alignment=right}]
                 Q[si={table-format=+1.2e3,table-number-alignment=right}]
                 Q[si={table-format=+1.2e3,table-number-alignment=right}]
             }
        }
        -5.25        & +0           & +5.25        \\
        -6.56919e-2  & -6.56919e-2  & -6.56919e-2  \\
        -1.18719e-2  & -1.18719e-2  & -1.18719e-2  \\
        +1.58439e-4  & +1.58439e-4  & +1.58439e-4  \\
        -7.56097e-7  & -7.56097e-7  & -7.56097e-7  \\
        +1.54262e-9  & +1.54262e-9  & +1.54262e-9  \\
        -1.11245e-12 & -1.11245e-12 & -1.11245e-12 \\
    \end{tblr}
\end{document}
```

Result is:

![image](https://github.com/lvjr/tabularray/assets/33900895/ebed1e07-65da-41dc-b17b-efdb77ab64be)

Is this something wrong with tabularray or I do something wrong? Please, help!

**Comment by muzimuzhi** [comment]
You need `table-format=+1.5e3`.

Or, check the `siunitx` option `table-auto-round`.

**Reply by muzimuzhi** [reply]
`table-format` sets the static space (width) reserved for all cells in this column, so longer number won't auto enlarge column width. If you don't want auto rounding, then a proper `table-format` which covers the longest number in this column is needed. Otherwise `table-auto-round` helps.

**Comment by sindzicat** [comment]
@muzimuzhi, thanks a lot! I didn't understand it correctly when reading siunitx documentation.
